src/lib/FlightTasks/CMakeLists.txt

```diff
@@ -31,23 +31,87 @@
 #
 ############################################################################
 
-px4_add_library(flight_tasks
-tasks/FlightTask.cpp
-		tasks/FlightTaskManual.cpp
-		tasks/FlightTaskManualStabilized.cpp
-		tasks/FlightTaskOrbit.cpp
-		tasks/FlightTaskManualAltitude.cpp
-		tasks/FlightTaskManualAltitudeSmooth.cpp
-		tasks/FlightTaskManualPosition.cpp
-		tasks/FlightTaskManualPositionSmooth.cpp
-		tasks/FlightTaskAuto.cpp
-		tasks/FlightTaskAutoLine.cpp
-		tasks/FlightTaskAutoMapper.cpp
-		tasks/FlightTaskAutoFollowMe.cpp
-		tasks/FlightTaskOffboard.cpp
-		tasks/Utility/ManualSmoothingZ.cpp
-		tasks/Utility/ManualSmoothingXY.cpp
-		tasks/Utility/StraightLine.cpp
-		SubscriptionArray.cpp
-		FlightTasks.cpp
+###########################################
+# Prepare flight tasks
+###########################################
+
+# add upstream flight tasks (they are being handled differently from the core below)
+list(APPEND flight_tasks_to_add
+	Orbit
 )
+# remove possible douplicates
+list(REMOVE_DUPLICATES flight_tasks_to_add)
+
+# remove flight tasks depending on target
+if(flight_tasks_to_remove)
```

Comment:
I think this `if` is not necessary. If `flight_tasks_to_add` is empty, then 
`list(REMOVE_ITEM flight_tasks_to_add
		${flight_tasks_to_remove}` will not do anything